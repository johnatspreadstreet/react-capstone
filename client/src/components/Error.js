import React, { Component } from 'react';

export default class Error extends Component {
  render() {
    const { error } = this.props;
    return (
      <div className="bg-red-lightest border-l-4 border-red text-red-dark p-2" role="alert">
        <p>{error}</p>
      </div>
    );
  }
}
