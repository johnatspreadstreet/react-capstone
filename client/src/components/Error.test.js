import React from 'react';
import { shallow, mount } from 'enzyme';
import Error from './Error';

describe('A suite of tests for the App component', () => {
  // Smoke test
  it('should render with no errors', () => {
    const component = shallow(<Error />);

    expect(component).toMatchSnapshot();
  });

  it('should render banner text correctly with given props', () => {
    const error = 'This is a test message';
    const component = shallow(<Error error={error} />);

    expect(component).toMatchSnapshot();
  });
});
