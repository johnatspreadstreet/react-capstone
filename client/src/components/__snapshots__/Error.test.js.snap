// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite of tests for the App component should render banner text correctly with given props 1`] = `
<div
  className="bg-red-lightest border-l-4 border-red text-red-dark p-2"
  role="alert"
>
  <p>
    This is a test message
  </p>
</div>
`;

exports[`A suite of tests for the App component should render with no errors 1`] = `
<div
  className="bg-red-lightest border-l-4 border-red text-red-dark p-2"
  role="alert"
>
  <p />
</div>
`;
